src/binmode/legacy4third.c

```diff
@@ -533,16 +549,66 @@ void legacy_protocol(void) {
                 if (binmode_debug) {
                     printf("\r\n>> ");
                 }
+                bool is_read_sig_cmd = false;  // true when reading any one of the signature bytes
+                uint8_t read_sig_byte_inx = 0; // which signature byte to read
                 for (int i = 0; i < bytes2read; i++) {
                     if (button_get(0)) {
                         return;
                     }
                     if (binmode_debug) {
                         printf("\r\n0x%02X | ", tmpbuf[i]);
                     }
+                    // This hack required to handle parts with more than
+                    // 64K bytes of flash.  Signature byte 2 indicates flash
+                    // size - we'll use this byte to set a global flag if
+                    // this part requires writing the Extended High Byte address
+                    // First - check if we are receiving a command 0x30 (query signature);
+                    // this command is 4 bytes of SPI:
+                    // 0 -> avrdude/BP sends 0x30
+                    // 1 -> avrdude/BP clocks out 0, and part returns 0x30 to confirm command
+                    // 2 -> avrdude/BP sends which of the 3 signature bytes to read
```

Comment:
`sends which of the 3 signature bytes to read` --> 
`sends which of the 3 signature byte addresses to read`  ... to more closely match the datasheet description of `Read Signature Byte`